src/runtime/store/index.ts

```diff
@@ -13,7 +13,7 @@ export type Updater<T> = (value: T) => T;
 type Invalidator<T> = (value?: T) => void;
 
 /** Start and stop notification callbacks. */
-export type StartStopNotifier<T> = (set: Subscriber<T>) => Unsubscriber | void;
+export type StartStopNotifier<T> = (set: Subscriber<T>, update?: (fn: Updater<T>) => void) => Unsubscriber | void;
```

Comment:
This needs to be a non-optional parameter, else TS will error here if you try to use `update` without checking for its existance first.
Which makes me think: Is this a breaking change for people implementing their own store contract? If so, yeah it needs to be an optional `update` parameter. @Conduitry what's your stance on this?

If we leave it like this, this means we need to adjust the other signatures a bit to type them so that `update` is _not_ optional as it is in fact provided for `readable`/`writable`/`derived`

Reply:
Strange, the two unit tests I added ran just fine, and I'm not seeing any TS errors in VS code on my use of `update` in the tests. I thought it *would* be a breaking change (at least from Typescript's perspective if not from Javascript's perspective) to make this non-optional as then everyone who has written `readable(0, (set) => { });` would get Typescript complaining that they didn't take the second parameter. But then, I don't know all the subtleties of Typescript; maybe it's fine with omitting parameters to functions because Javascript allows that?

Reply:
BTW, at no point in the Svelte code does the `update` function from StartStopNotifier get called; it's only passed to callbacks. And the callbacks can either take the `update` parameter and use it, or not take it in which case it would obviously be an error for them to have a call to `update(x => x+1)`. But I don't think making this optional will actually produce any Typescript errors, and I do think that making it non-optional will be a breaking change.

Reply:
For function parameters, the inverse is true: If the function expects a function-parameter with x arguments, you can pass in a function with zero to x parameters.
You don't see errors because the repository runs TS in non-strict mode (no strict null checks).
See this playground for more: https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABACwKYBt1wBTDALkWwCdUBHQgZymJjAHMAaROABygH4qa76BKRAF4AfIgBucGABMBAbwC+AKEVpMObAJGIFfANyIA9AcRg4iVsTgAjdKgC2KjFmwlyzNlE2jZixH5bsAHRQcAAycADuqMQAwgCGlKga+kaI0ZbEiFaoEHEgiQFQiBBwIOhSWaiI4FKowHSoUoryfEA

Reply:
I see. And https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABACwKYBt1wBTDALkWwCdUBHQgZymJjAHMAaROAByiprvoEpEBeAHyIAbnBgATPgG8AvgCh5aTDmx8hiOTwDciAPR7EYOIlbE4AI3SoAtkoxZsJcuuHT9huDZhRYDFuyIMJRGJmaW1jaIqD5oxPKIiYikZAB0UHAAMnAA7qjEAMIAhpSoarrysjzyQA shows that omitting `update` will be no problem, so it won't break existing code. I'll make the change.

Reply:
It's a breaking change for everyone who uses `StartStopNotifier` to implement their own store contract: https://www.typescriptlang.org/play?#code/C4TwDgpgBAysCGAnYcD2YByrgEsBmOEiUAvFABQDOEwAXFJcIjgHYDmANFAK5gAm8YBHqNm7AJSkAfFABuqHHwDcAKBV5uLAMa5ULKAFsQAeQDuLNIgjkAFhAA291PXgsQXRtzx56cJCmB0LFwCIkkAbxUoaIZgLzxyAHJUez5E8SUoAHosqDArWRxUbkp7ECgCFgguFlRTKCJEVERKKAAjCC14EuheASEoHFba+vgoKwBHbhwrPjykeAMaIhUAXyA